### PR TITLE
[CM-1341] Added Support For Auto Suggestions Rich Message

### PR DESCRIPTION
**Pull request by AbhijeetRanjan308**
## Summary
- Added Support of the Auto Suggestion Rich Message 
- The chat bar placeholder is updated when the rich message is called.
- Used Auto Complete Function for this. Created few methods and added a condition to check for autoSuggestion.
- Created function to convert data to array or dictionary as per need.
- Created function for api call to set the data in dictionary.
- autoSuggestion work after the two key words.
- api call is happening after ever keyword. And refreshing the search list every time.

https://github.com/Kommunicate-io/KommunicateChatUI-iOS-SDK/assets/139110221/48a70cb2-c150-48a6-8748-e70790195add

### CODE REVIEW

File: ALKRichMessageAutoSuggestion.swift [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        }
+        
+        let session = URLSession.shared
+        let task = session.dataTask(with: url) { (data, _, error) in
+            if let error = error {
+                print("Error: \(error)")
+                return
+            }
+            guard let data = data, let dataString = String(data: data, encoding: .utf8), let jsonData = dataString.data(using: .utf8) else {
+                print("Invalid data")
```
Comment (Sathyan-KM) [inline ALKRichMessageAutoSuggestion.swift:56]: need clear print statement here

File: ALKRichMessageAutoSuggestion.swift [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    func fetchData(from url: String) {
+        guard let url = URL(string: url) else {
+            print("Invalid URL")
+            return
+        }
+        
+        let session = URLSession.shared
+        let task = session.dataTask(with: url) { (data, _, error) in
+            if let error = error {
+                print("Error: \(error)")
```
Comment (Sathyan-KM) [inline ALKRichMessageAutoSuggestion.swift:52]: need clear print statement here..

File: ALKRichMessageAutoSuggestion.swift [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            guard let data = data, let dataString = String(data: data, encoding: .utf8), let jsonData = dataString.data(using: .utf8) else {
+                print("Invalid data")
+                return
+            }
+            do {
+                if let jsonObject = try JSONSerialization.jsonObject(with: jsonData) as? [String: Any], let dataArray = jsonObject["data"] as? [[String: Any]] {
+                    self.suggestionDict = dataArray
+                }
+            } catch {
+                print("Error parsing JSON: \(error)")
```
Comment (Sathyan-KM) [inline ALKRichMessageAutoSuggestion.swift:64]: same logs issue

File: ALKRichMessageAutoSuggestion.swift (hunk 2 above)
Comment (AbhijeetRanjan308) [reply ALKRichMessageAutoSuggestion.swift:52]: okay I will make the logs clear

File: ALKRichMessageAutoSuggestion.swift (hunk 3 above)
Comment (AbhijeetRanjan308) [reply ALKRichMessageAutoSuggestion.swift]: updated the comments

File: ALKRichMessageAutoSuggestion.swift (hunk 1 above)
Comment (AbhijeetRanjan308) [reply ALKRichMessageAutoSuggestion.swift]: changed the comments

File: Sources/Controllers/ALKConversationViewController.swift [hunk 4]
```diff
@@ -2318,6 +2355,11 @@ extension ALKConversationViewController: ALKConversationViewModelDelegate {
         // Clear reply message and the view
         viewModel.clearSelectedMessageToReply()
         hideReplyMessageView()
+        autoSuggestionView.isHidden = true
+        isAutoSuggestionRichMessage = false
+        suggestionDict = []
+        suggestionArray = []
```
[the remark this answers is not in the document]
Comment (Sathyan-KM) [reply Sources/Controllers/ALKConversationViewController.swift:2362]: @AbhijeetRanjan308  ?

[the remark this answers is not in the document]
Comment (AbhijeetRanjan308) [reply Sources/Controllers/ALKConversationViewController.swift]: done

**Review by Sathyan-KM — APPROVED** [review]

**Pull request merged**